nilearn/datasets/tests/test_utils.py

```diff
@@ -188,37 +187,34 @@ def test_get_dataset_dir_symlink(tmp_path):
     )
 
     assert data_dir == str(expected_linked_dir)
-    assert os.path.exists(data_dir)
+    assert Path(data_dir).exists()
 
 
-def test_md5_sum_file():
+def test_md5_sum_file(tmp_path):
     # Create dummy temporary file
-    out, f = mkstemp()
-    os.write(out, b"abcfeg")
-    os.close(out)
+    f = tmp_path / "test"
+    f.write_bytes(b"abcfeg")
 
     assert _utils._md5_sum_file(f) == "18f32295c556b2a1a3a8e68fe1ad40f7"
 
-    os.remove(f)
+    f.unlink()
```

Comment:
```suggestion
```

because we are using the `tmp_path` fixture I don't think we need to manually handle all the teardown part of the test